BurnOutSharp/ProtectionType/CDGuard.cs

```diff
@@ -0,0 +1,90 @@
+    /// There is a re-release of an earlier game by the same developer that seems to include both CD-Guard and StarForce drivers, with the CD-Guard driver seemingly not used during installation, nor installed onto the system (IA item "pahgeby-he3hakomkou").
+    /// 
+    /// Additional resources and references:
+    /// https://gamecopyworld.com/games/pc_omikron.shtml
+    /// https://forum.ixbt.com/topic.cgi?id=31:3985
+    /// </summary>
+    public class CDGuard : IPathCheck, IPortableExecutableCheck
+    {
+        /// <inheritdoc/>
+        public string CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+        {
+            // Get the sections from the executable, if possible
+            var sections = pex?.SectionTable;
+            if (sections == null)
+                return null;
+
+            // TODO: Investigate the numerous ".guard" sections present in "Randevu.exe" in Redump entry 97142.
+
+            // Get the export directory table
+            if (pex.ExportTable?.ExportDirectoryTable != null)
+            {
+                // Found in "cdguard.dll" in Redump entry 97142.
+                bool match = pex.ExportTable.ExportDirectoryTable.Name == "CDGUARD.DLL";
+                if (match)
+                    return "CD-Guard Copy Protection System";
+
+                // Found in "cdguard.dll" in IA item "pahgeby-he3hakomkou".
+                match = pex.ExportTable.ExportDirectoryTable.Name == "cdguard.dll";
+                if (match)
+                    return "CD-Guard Copy Protection System";
+            }
+
+            // Get the import directory table
+            if (pex.ImportTable?.ImportDirectoryTable != null)
+            {
+                // Found in "Randevu.exe" in Redump entry 97142.
+                bool match = pex.ImportTable.ImportDirectoryTable.Any(idte => idte.Name == "CDGUARD.DLL");
+                if (match)
+                      return "CD-Guard Copy Protection System";
+
+                // Not found in any known sample, but presumably would be found in the game EXE for a game that uses the same driver version as IA item "pahgeby-he3hakomkou".
+                match = pex.ExportTable.ExportDirectoryTable.Name == "cdguard.dll";
+                if (match)
+                    return "CD-Guard Copy Protection System";
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string> files)
+        {
+            var matchers = new List<PathMatchSet>
+            {
+                // Found in Redump entry 97142.
+                new PathMatchSet(new PathMatch("cdguard.dll", useEndsWith: true), "CD-Guard Copy Protection System"),
```

Comment:
Do you think it's necessary to put the directory separator in front of those? Or is it unlikely to get false positives?

Reply:
I'm not worried about that atm personally, but I can add it if you'd like.